idaes/core/util/tests/test_model_diagnostics.py

```diff
@@ -527,6 +596,30 @@ def model(self):
 
         return m
 
+    @pytest.mark.component
+    def test_with_grey_box(self):
+
+        class BasicGrayBox(ExternalGreyBoxModel):
+            def input_names(self):
+                return ["a1", "a2", "a3"]
+
+            def output_names(self):
+                return ["o1", "o2"]
+
+            def equality_constraint_names(self):
+                return ["a_sum"]
+
+            def evaluate_equality_constraints(self):
+                a1 = self._input_values[0]
+                a2 = self._input_values[1]
+                return [a1 * 0.5 + a2]
```

Comment:
Just wondering if this needs to be replicated within tests like you do now or if you could just create once outside of tests.